pkg/eventqueue/eventqueue.go

```diff
@@ -121,6 +138,12 @@ type eventStatistics struct {
 	waitConsumeOffQueue spanstat.SpanStat
 }
 
+// UpdateName updates the name of the EventQueue.
+func (q *EventQueue) UpdateName(name string) {
```

Comment:
A name of `EventQueue` should never change during its execution, a name should be part of the struct's identity. If you really really want the ability to change the name then you need to protect it against concurrent access.

Reply:
I'll add a mutex which protects this field. The reason this has to be done is because I want to name the queue based off of the endpoint ID, but the endpoint ID is not allocated as part of endpoint creation - it is done by the endpointmanager.

Reply:
Added a mutex to protect the field in a follow-up commit.

Reply:
How much work is it to shift the endpointID allocation to before the endpoint initialization? It seems strange to me that something so fundamental to the endpoint is actually considered mutable state that's being modified after the endpoint object is initialized.

Reply:
🤔 I checked out the code and I see no reason why we can't move the endpoint ID allocation to endpoint creation. I'll play around with that.

Reply:
Be advised that the main path is:
```
func (d *Daemon) createEndpoint(ctx context.Context, epTemplate *models.EndpointChangeRequest) (*endpoint.Endpoint, error) {

        ep, err := endpoint.NewEndpointFromChangeModel(epTemplate)
```

The endpoint ID will *not* be set in the template. It is assigned on insertion into the manager. The plugin calling the API is no longer responsible to allocate the endpoint ID. Allocating earlier is possible but it will complicate the endpoint ID release mechanism in all of the error paths.

Reply:
@tgraf agreed, I came to that conclusion eventually as well. The correct solution is to start the queue as the endpoint is inserted into the endpointmanager (i.e., after it has gotten an ID).

Reply:
That sounds like a fair compromise.